astropy/io/fits/tests/test_header.py

```diff
@@ -617,12 +617,9 @@ def test_missing_keyword(self):
         """Test that accessing a non-existent keyword raises a KeyError."""
 
         header = fits.Header()
-        pytest.raises(KeyError, lambda k: header[k], 'NAXIS')
-        # Test the exception message
-        try:
+        # Test exception with message
+        with pytest.raises(KeyError, match="Keyword 'NAXIS' not found."):
```

Comment:
Since the `match` is a regex, the `.` should probably be removed because it matches anything not just a dot, right?

Reply:
Same for the other changes here that contain `.`

Reply:
At least it asserts that the period is not missing entirely, but I could not identify any common standard for completeness or uniqueness in the `io.fits` tests, so I guess anything goes.

Reply:
Could you escape the dots to make sure that it doesn't match another character?

Reply:
at a lot of other places we only match for parts of the error message. I suppose there aren't two cases, one with period and one without, so dropping a period would in the test above would serve the aim of the test perfectly.